services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -175,23 +174,22 @@ async def list_files(  # noqa C901
                 sha256_checksum=None,
             )
 
-            # add all the entries from file_meta_data without
-            for metadata in file_and_directory_meta_data:
-                # below checks ensures that directoris either appear as
-                if metadata.is_directory and expand_dirs:
-                    # avoids directory files and does not add any directory entry to the result
-                    continue
-
-                if is_file_entry_valid(metadata):
-                    data.append(convert_db_to_model(metadata))
-                    continue
-                with suppress(S3KeyNotFoundError):
-                    updated_fmd = await self._update_database_from_storage(
-                        conn, metadata
-                    )
-                    data.append(convert_db_to_model(updated_fmd))
+        # add all the entries from file_meta_data without
+        for metadata in file_and_directory_meta_data:
+            # below checks ensures that directoris either appear as
+            if metadata.is_directory and expand_dirs:
+                # avoids directory files and does not add any directory entry to the result
+                continue
 
-            # now parse the project to search for node/project names
+            if is_file_entry_valid(metadata):
+                data.append(convert_db_to_model(metadata))
+                continue
+            with suppress(S3KeyNotFoundError):
+                updated_fmd = await self._update_database_from_storage(metadata)
+                data.append(convert_db_to_model(updated_fmd))
+
+        # now parse the project to search for node/project names
+        async with self.engine.acquire() as conn:
```

Comment:
So this is the most critical change, namely that you release the engine and re-acquire it  only when you need to use it.

Reply:
yep